system/HTTP/CLIRequest.php

```diff
@@ -214,4 +214,60 @@ public function isCLI(): bool
     {
         return true;
     }
+
+    /**
+     * Fetch an item from GET data.
+     *
+     * @param array|string|null $index  Index for item to fetch from $_GET.
+     * @param int|null          $filter A filter name to apply.
+     * @param mixed|null        $flags
+     *
+     * @return null
+     */
+    public function getGet($index = null, $filter = null, $flags = null)
+    {
+        return null;
```

Comment:
I'm not sure about this. I worry that it will hide mistakes. Have you considered throwing an exception? `BadMethodCallException` maybe?

Reply:
Now I think `CLIRequest` should have all the methods that `IncomingRequest` has.

The following code does not work in CLI in 4.2.4. But I'm not sure the code should raise an error.
At least in CI3 or CI4.1 it runs without an error.

```php
<?php
namespace App\Controllers;
class Home extends BaseController
{
    public function index()
    {
        $this->request->getPost();
    }
}
```

It is true that there is no POST data in CLI, so you should not use `getPost()`.

But we do not know what legacy codes users have.
And It doesn't seem good to have code that used to work without error stop working.

I'm afraid that users will continue to report bugs that the code no longer works when it used to work.

Reply:
If the goal is code compatibility shouldn't it be something like this?
```php
return $index === null ? [] : null;
```

Reply:
Fixed.